### PR TITLE
add invicible in increase_attack

### PR DESCRIPTION
**Pull request by UbiquinoneUQ**

### CODE REVIEW

File: Classes/core/entity/entity.cpp [hunk 1]
```diff
@@ -11,6 +11,9 @@ void Entity::on_tick(GridRef g) {}
 void Entity::on_hit(GridRef g) {}
 
 void Enemy::increase_attack(int32_t atk, AttackType attack_type) {
+    if(get_all_buff().invincible_){
```
Comment (snylonue) [inline Classes/core/entity/entity.cpp:14]: what if atk < 0 (aka is healing)

File: Classes/core/entity/entity.cpp [hunk 2]
```diff
@@ -36,14 +39,22 @@ void Enemy::on_tick(GridRef g) {
             auto ny = route::ssize(g.column) + dy;
             assert(nx >= 0 && nx < g.map.shape.height_ && ny >= 0 && ny < g.map.shape.width_);
             g.map.move_enemy_to(this->id, nx, ny);
-            g.on_enemy_move(*this, std::make_pair(route::ssize(g.row), route::ssize(g.column)), std::make_pair(nx, ny));
+            if(dx < -1||dx > 1||dy < -1||dy >1){
```
Comment (snylonue) [inline Classes/core/entity/entity.cpp:42]: Why add an `on_transport`?

**Review by snylonue — CHANGES_REQUESTED** [review]

File: Classes/core/entity/entity.cpp (hunk 2 above)
Comment (UbiquinoneUQ) [reply Classes/core/entity/entity.cpp:42]: animation of transport is different from move

Comment (UbiquinoneUQ) [reply Classes/core/entity/entity.cpp]: deleted

Comment (snylonue) [reply Classes/core/entity/entity.cpp]: > animation of transport is different from move

Then it's ok to add it. Though I perfer providing a function (of the two coordinates) to tell whether it is transport or regular move or adding a `bool is_transport` in callback.

**Review by snylonue — APPROVED** [review]

**Pull request merged**